worker/src/main/java/org/apache/celeborn/service/deploy/worker/WriterHelper.java

```diff
@@ -0,0 +1,119 @@
+    this.storageManager = storageManager;
+    this.partitionLocationInfo = partitionLocationInfo;
+  }
+
+  public List<PartitionLocation> createWriters(
+      String shuffleKey,
+      String applicationId,
+      int shuffleId,
+      List<PartitionLocation> requestLocs,
+      long splitThreshold,
+      PartitionSplitMode splitMode,
+      PartitionType partitionType,
+      boolean rangeReadFilter,
+      UserIdentifier userIdentifier,
+      boolean partitionSplitEnabled,
+      boolean isSegmentGranularityVisible,
+      boolean isPrimary) {
+    List<PartitionLocation> locs = new ArrayList<>();
+    try {
+      ConcurrentLinkedQueue<PartitionLocation> locQueue = new ConcurrentLinkedQueue<>();
+      requestLocs
+          .parallelStream()
+          .forEach(
+              requestLoc -> {
+                PartitionLocation location =
+                    isPrimary
+                        ? partitionLocationInfo.getPrimaryLocation(
+                            shuffleKey, requestLoc.getUniqueId())
+                        : partitionLocationInfo.getReplicaLocation(
+                            shuffleKey, requestLoc.getUniqueId());
+                if (location == null) {
+                  location = requestLoc;
+                  PartitionDataWriter writer;
+                  try {
+                    writer =
+                        storageManager.createPartitionDataWriter(
+                            applicationId,
+                            shuffleId,
+                            location,
+                            splitThreshold,
+                            splitMode,
+                            partitionType,
+                            rangeReadFilter,
+                            userIdentifier,
+                            partitionSplitEnabled,
+                            isSegmentGranularityVisible);
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
+                  }
+                  locQueue.add(new WorkingPartition(location, writer));
+                } else {
+                  locQueue.add(location);
+                }
+              });
+      locs.addAll(locQueue);
+    } catch (Exception e) {
+      LOG.error("CreateWriter for {} failed.", shuffleKey, e);
+    }
+    return locs;
+  }
+
+  public void destroyWriters(List<PartitionLocation> locs, String shuffleKey) {
+    locs.parallelStream()
+        .forEach(
+            partitionLocation -> {
+              WorkingPartition workingPartition = (WorkingPartition) partitionLocation;
+              PartitionDataWriter fileWriter = workingPartition.getFileWriter();
+              fileWriter.destroy(
+                  new IOException(
+                      String.format(
+                          "Destroy FileWriter %s caused by " + "reserving slots failed for %s.",
+                          fileWriter, shuffleKey)));
+            });
+  }
+}
```

Comment:
not sure how much benefit with parallelStream.

Some concern:
1. for `parallelStream`, the parallelize is determined by the system processor number 
2. and `parallelStream` is for CPU intensive, for here, is it IO intensive?

Reply:
maybe use `newDaemonFixedThreadPool`?